scip.proto

```diff
@@ -383,10 +383,31 @@ message Occurrence {
   // type with `start` and `end` fields of type `Position`, mirroring LSP.
   // Benchmarks revealed that this encoding was inefficient and that we could
   // reduce the total payload size of an index by 50% by using `repeated int32`
-  // instead.  The `repeated int32` encoding is admittedly more embarrassing to
+  // instead. The `repeated int32` encoding is admittedly more embarrassing to
   // work with in some programming languages but we hope the performance
   // improvements make up for it.
   repeated int32 range = 1;
+  // (optional) The source position of the nearest non-trivial enclosing AST node.
+  // If supplied, the value must mirror the constraints detailed in the `range` field.
+  // For occurrences referencing the identifier within a class or function declaration, 
```

Comment:
I usually say "definition occurrence" to refer to occurrences that have the definition role
```suggestion
  // For definition occurrences, the enclosing range should encode the start/end bounds of the entire definition AST node.
```